packages/utils/src/tableUtils/index.js

```diff
@@ -0,0 +1,73 @@
+export const objectHasValueForKeys = (item = {}, keys = []) => Object.entries(item).reduce(
+  (hasValue, [key, value]) => hasValue || (keys.includes(key) && (value !== undefined)), false,
+);
+
+export const arrayHasValueForKeys = (array = [], keys = []) => ((array.length > 0) && array.reduce(
+  (hasValue, item) => hasValue || objectHasValueForKeys(item, keys), false,
+));
+
+export function keysFromFields(fields) {
+  return fields.map(field => field.key);
+}
+
+export const titlesFromFields = fields => fields.map(field => field.heading || field.title);
+
+export const rowsFromArray = (array, fields, skipEmptyRows) => {
+  const keys = keysFromFields(fields);
+
+  return array.reduce((rows, item) => {
+    if (!skipEmptyRows || objectHasValueForKeys(item, keys)) {
+      rows.push(
+        fields.map(({ key, transform = result => (result || '-')}) => transform(item[key], item)),
+      );
+    }
+
+    return rows;
+  }, []);
+};
+
+// TODO: ALL THE DOCS
+// TODO: SOME TESTS
+// Empty values and keys are treated the same
+export const rowsFromObject = (object, fields, skipEmptyValues, defaultTransform) => {
```

Comment:
i'd have been inclined to have ```defaultTransform = value => `${value}` ``` here
which would let line 34 below be
```
(table, { key, heading, names, transform = defaultTransform }) => {
```
and then allow lines 41-57 to be replaced with just:
```
const result = transform(object[key], object) || '';
```

Reply:
I've made this change, great spot, thanks!